lib/util/reddit.ts

```diff
@@ -0,0 +1,68 @@
+import * as centra from "centra";
+import { randomInt } from "crypto";
+import { RedditPost } from "../interfaces/reddit";
+
+/* API */
+
+/**
+ * Returns a random post/meme from a subreddit.
+ * @param subreddit Optional The subreddit string, defaults to a random english meme subreddit.
+ * @returns {RedditPost}
+ */
+ export async function getRandomPost(subreddit?: string): Promise<RedditPost | undefined> {
+	const url = `https://www.reddit.com/r/${subreddit ? subreddit : randomSubreddit()}/hot/.json?count=100`
+
+	return await buildPost(url)
+}
+
+
+/* Utils */
+
+// more languages coming soontm
+const randomSubreddits = {
+  en: ['memes', 'dankmemes', 'meirl'],
+}
+
+/**
+ * Gets a random subreddit from the list of available meme subreddits.
+ * @returns {string} The random subreddit string.
+ */
+function randomSubreddit(): string {
```

Comment:
you should make this accept a language

Reply:
it will when i have more than one lenguage

Reply:
better to implement now than to forget later :)

Reply:
done